src/main/java/com/echobox/api/linkedin/client/DefaultWebRequestor.java

```diff
@@ -82,361 +55,145 @@ public class DefaultWebRequestor implements WebRequestor {
-      }
-
-      return getResponse(request);
-    } catch (HttpResponseException ex) {
-      return handleException(ex);
-    } finally {
-      if (autocloseBinaryAttachmentStream && binaryAttachments.length > 0) {
-        for (BinaryAttachment binaryAttachment : binaryAttachments) {
-          closeQuietly(binaryAttachment.getDataInputStream());
-        }
-      }
+  private HttpRequest.BodyPublisher buildJsonRequestBody(String jsonBody,
+      HttpRequest.Builder builder) {
+    if (jsonBody != null) {
+      builder.header("Content-Type", "application/json");
+      return HttpRequest.BodyPublishers.ofString(jsonBody);
     }
+    return HttpRequest.BodyPublishers.noBody();
   }
   
-  @Override
-  public Response executePut(String url, String parameters, String jsonBody,
-      Map<String, String> headers, BinaryAttachment binaryAttachment)
-      throws IOException {
-  
+  private HttpRequest.BodyPublisher buildRequestBodyWithBinaryAttachments(String url,
+      HttpRequest.Builder builder,
+      Map<String, String> headers, BinaryAttachment... binaryAttachments) {
     try {
-      GenericUrl genericUrl = getGenericURL(url, parameters);
-    
-      HttpRequest request;
-      HttpHeaders httpHeaders = new HttpHeaders();
-    
-      // If we have binary attachments, the body is just the attachments and the
-      // other parameters are passed in via the URL.
-      // Otherwise the body is the URL parameter string.
-      if (binaryAttachment != null) {
-        // Set the content type
-        // If it's not provided assume it's application/octet-stream
-        String contentType =
-            headers.entrySet().stream()
-                .filter(entry -> entry.getKey().equalsIgnoreCase("content-type"))
-                .map(Map.Entry::getValue).findFirst().orElse("application/octet-stream");
-        ByteArrayContent fileContent =
-            new ByteArrayContent(contentType, binaryAttachment.getData());
-        request = requestFactory.buildPutRequest(genericUrl, fileContent);
-        httpHeaders.set("Connection", "Keep-Alive");
-      } else {
-        if (jsonBody != null) {
-          request = requestFactory.buildPutRequest(genericUrl, getJsonHttpContent(jsonBody));
-        
-          // Ensure the response headers are also set to JSON
-          request.setResponseHeaders(new HttpHeaders().set(FORMAT_HEADER, "json"));
-        } else {
-          // Plain old PUT request
-          request = requestFactory.buildPutRequest(genericUrl, null);
-        }
+      if (binaryAttachments.length == 1) {
+        String contentType = headers.entrySet().stream()
+            .filter(entry -> entry.getKey().equalsIgnoreCase("content-type"))
+            .map(Map.Entry::getValue).findFirst().orElse("application/octet-stream");
+        builder.header("Content-Type", contentType);
+        return HttpRequest.BodyPublishers.ofByteArray(binaryAttachments[0].getData());
```

Comment:
Not having this separated out if it was 1 binary attachment was what was causing issues previously. Though I would have thought that it would work with just one 1 file anyway. This has now fixed issues.

Though we don't seem to have a case in the platform where there are multiple binary attachments in the same request (I think) for linkedin so I can confidently test that.

Reply:
Yeah I have checked and it is actually the `VideoConnection` code in this repo that sets the number of binary attatchments....I think it is not worth testing further at this point